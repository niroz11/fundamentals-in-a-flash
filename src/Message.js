import React, { Component } from 'react';

class Messsage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  updateUserStatus = (event) => {
    event.preventDefault();
    this.props.updateUserStatus('finished')
  }

  render() {
    let { userStatus, currentQuestion, updateUserStatus, moveToNextQuestion, isFinalQuestion } = this.props;
    let { id, category, question, answers, correct_answer, resources } = currentQuestion;
    let style = category.toLowerCase().split(' ').join('-')
    return (
      <div className={"message-contain " + style}>
      {
        userStatus === 'correct' && 
          <p>Yay! <span className="answer">{correct_answer}</span> is the correct answer</p>
      }
      {
        userStatus === 'wrong' && 
          <p><span className="sorry">Sorry that is not correct.</span>Check out the <a href={resources} target="_blank">docs</a> for more information</p>
      }
      {
          !isFinalQuestion && <button className="next-btn" onClick={moveToNextQuestion}>Next</button>
      }
      {
          isFinalQuestion &&  <button className="finish-btn" onClick={this.updateUserStatus}>Finish</button>
      }
      </div>
    )
  }
}

export default Messsage;