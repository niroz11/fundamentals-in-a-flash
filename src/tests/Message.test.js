import React from 'react';
import Message from '../Message';
import { shallow } from 'enzyme';

const userStatus = 'correct';
const currentQuestion = {
  id: "17",
  category: "OOP",
  question: "When creating a new class that extends from a parent class, what keyword is used to access and invoke methods from the parent class?",
  answers: [ "function", "borrow", "super"],
  correct_answer: "super",
  resources: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super"
}
const updateUserStatusMock = jest.fn();
const moveToNextQuestionMock = jest.fn();

describe('Message', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Message
        userStatus={userStatus} 
        currentQuestion={currentQuestion} 
        updateUserStatus={updateUserStatusMock} 
        moveToNextQuestion={moveToNextQuestionMock} 
        isFinalQuestion={false}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should move to the next question', () => {
    wrapper.find('button.next-btn').simulate('click');
    expect(moveToNextQuestionMock).toHaveBeenCalled();
  });

  it('should update state to finished', () => {
    wrapper.instance().updateUserStatus({ preventDefault: () => { }, target: { innerHTML: 'Finish' }});
    expect(updateUserStatusMock).toHaveBeenCalledWith('finished');
  });

  it('should update state to guessing when the user is wrong and clicks try again', () => {
    wrapper.instance().updateUserStatus({ preventDefault: () => { }, target: { innerHTML: 'Try Again' } });
    expect(updateUserStatusMock).toHaveBeenCalledWith('guessing');
  });
});