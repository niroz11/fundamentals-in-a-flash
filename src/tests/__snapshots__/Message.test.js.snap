// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match the snapshot with all data passed in 1`] = `
<div
  className="message-contain oop"
>
  <p>
    Yay! 
    <span
      className="answer"
    >
      super
    </span>
     is the correct answer
  </p>
  <button
    className="next-btn"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
